### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by emstoudenmire**
This PR makes the following changes to the README file:

1. fix a mistake at the beginning which refers to named edges, whereas it should be named vertices, correct?

2. add a sentence explaining that the names could be strings, tuples of integers, or other things

3. remove some extraneous empty lines

### CODE REVIEW

**Comment by mtfishman** [comment]
The README is actually generated from this file: https://github.com/ITensor/NamedGraphs.jl/blob/main/examples/README.jl using the instructions at the bottom of that file, can you make the edits to that file and then generate the new README?

**Comment by mtfishman** [comment]
Of course we should automate that process in a CI workflow but I haven't gotten around to that yet.

**Comment by mtfishman** [comment]
(That's why there is the weird spacing, that's just how it gets generated.)

**Comment by emstoudenmire** [comment]
Sure thing – I just edited the `examples/README.jl` file and used the command at the bottom of it to regenerate the main README.jl

File: examples/README.jl [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 #' ## Introduction
 
-#' This packages introduces graph types with named edges, which are built on top of the `Graph`/`SimpleGraph` type in the [Graphs.jl](https://github.com/JuliaGraphs/Graphs.jl) package that only have contiguous integer edges (i.e. linear indexing).
+#' This packages introduces graph types with named vertices, which are built on top of the `Graph`/`SimpleGraph` type in the [Graphs.jl](https://github.com/JuliaGraphs/Graphs.jl) package that only have contiguous integer vertices (i.e. linear indexing). The vertex names can be strings, tuples of integers, or other unique identifiers. The vertex names can be strings, tuples of integers, or other unique identifiers.
```
Comment (mtfishman) [inline examples/README.jl:18]:
```suggestion
#' This packages introduces graph types with named vertices, which are built on top of the `Graph`/`SimpleGraph` type in the [Graphs.jl](https://github.com/JuliaGraphs/Graphs.jl) package that only have contiguous integer vertices (i.e. linear indexing). The vertex names can be strings, tuples of integers, or other unique identifiers (anything that is hashable).
```

Comment (mtfishman) [reply examples/README.jl:18]: (The last sentence was repeated.)

Comment (emstoudenmire) [reply examples/README.jl]: Whoops, fixed

**Comment by mtfishman** [comment]
Thanks! No idea why it was referring to "edges" instead of "vertices"...

**Comment by emstoudenmire** [comment]
Maybe you were working on the `NamedEdge` type write before you wrote that? 😄

**Pull request merged**